src/App.jsx

```diff
@@ -14,33 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function preparedGoodsData(goods, sortValue, isReversed) {
+  const preparedData = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortValue) {
+    preparedData.sort((good1, good2) => {
+      switch (sortValue) {
+        case SORT_BY_NAME:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedData.reverse();
+  }
+
+  return preparedData;
+}
+
+export const App = () => {
+  const [sortValue, setSortValue] = useState('');
+  const [isReversed, setReversed] = useState(false);
+
+  const visibleGoods = preparedGoodsData(
+    goodsFromServer,
+    sortValue,
+    isReversed,
+  );
+
+  const sortState = sortValue || isReversed;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  function handleReverse() {
+    setReversed(yes => !yes);
+  }
+
+  function handleReset() {
+    setSortValue('');
+    setReversed(false);
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortValue(SORT_BY_NAME)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortValue !== SORT_BY_NAME,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortValue(SORT_BY_LENGTH)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortValue !== SORT_BY_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => handleReverse()}
```

Comment:
using `onClick={() => handleReset()}` is fine if you need to pass arguments or perform additional logic, but for basic cases, `onClick={handleReset}` is more efficient and clean.

```suggestion
          onClick={handleReverse}
```